app/src/test/java/com/codesoom/assignment/application/UserServiceTest.java

```diff
@@ -0,0 +1,230 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.UserNotFoundException;
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DataJpaTest
+@DisplayName("UserService")
+public class UserServiceTest {
+    private static final String EXISTED_USER_NAME = "홍길동";
+    private static final String EXISTED_USER_EMAIL = "hong@gmail.com";
+    private static final String EXISTED_USER_PASSWORD = "password";
+    private static final Long NOT_EXISTED_USER_ID = 0L;
+
+    User exitedUser;
+    private UserService userService;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @BeforeEach
+    void prepareUserService() {
+        userRepository.deleteAll();
+        userService = new UserService(userRepository);
+    }
+
+    void prepareExitedUser() {
+        User user = User.builder()
+                .name(EXISTED_USER_NAME)
+                .email(EXISTED_USER_EMAIL)
+                .password(EXISTED_USER_PASSWORD)
+                .build();
+        exitedUser = userRepository.save(user);
+    }
+
+    @Nested
+    @DisplayName("findAllUsers는")
+    class Describe_findAllUsers {
+        List<User> subject() {
+            return userService.findAllUsers();
+        }
+
+        @Nested
+        @DisplayName("등록된 유저가 있다면")
+        class Context_with_existed_user {
+            @BeforeEach
+            void prepare() {
+                prepareExitedUser();
+            }
```

Comment:
너무너무 잘 하고 계세요. 그러니 다음 단계로도 더 올라가 봅시다!

```suggestion
        @Nested
        @DisplayName("등록된 유저가 한 명 있다면")
        class Context_with_existed_user {
            @BeforeEach
            void prepare() {
                prepareExitedUser();
            }
```

이렇게 바꾸고, 다음 컨텍스트들도 추가해 보세요.

- `등록된 유저가 2명 이상이라면`
- `등록된 유저가 아무도 없다면`